tests/install_upgrade_operators/console_cli_download/utils.py

```diff
@@ -0,0 +1,80 @@
+import logging
+
+from _pytest._py.path import LocalPath
+from kubernetes.dynamic import DynamicClient
+from timeout_sampler import TimeoutExpiredError, TimeoutSampler
+
+from utilities.constants import TIMEOUT_1MIN, TIMEOUT_5SEC, VIRTCTL_CLI_DOWNLOADS
+from utilities.infra import download_and_extract_file_from_cluster, get_console_spec_links
+
+LOGGER = logging.getLogger(__name__)
+CUSTOMIZED_VIRT_DL = "customized-virt-dl"
+
+
+def validate_custom_cli_downloads_urls_updated(
+    admin_client: DynamicClient,
+    new_hostname: str | None = None,
+    original_virtctl_console_cli_downloads_spec_links: list[str] | None = None,
+) -> None:
+    """
+    Validates that console CLI download URLs have been properly updated.
+
+    This function monitors the console CLI download links and validates one of two scenarios:
+    1. URLs have been reverted to their original state (rollback validation)
+    2. URLs have been updated to use a new hostname (update validation)
+
+    Args:
+        admin_client (DynamicClient): Kubernetes dynamic client for API operations
+        new_hostname (str, optional): New hostname that should be present in all URLs.
+            If provided, validates that all URLs contain this hostname.
+        original_virtctl_console_cli_downloads_spec_links (list[str], optional):
+            Original list of CLI download URLs. If provided, validates that current
+            URLs match these original URLs.
+
+    Returns:
+        None: Function returns when validation succeeds
+
+    Raises:
+        TimeoutExpiredError: If validation fails within the timeout period (1 minute).
+            This can occur when:
+            - URLs don't revert to original state within timeout
+            - URLs don't get updated with new hostname within timeout
+
+    Note:
+        - Exactly one of new_hostname or original_virtctl_console_cli_downloads_spec_links
+          should be provided, not both
```

Comment:
should be under args, mentioned for each arg, not here. 
the function does not enforce this rule; if this is the condition, please add a check in the function